storage/uploader.go

```diff
@@ -0,0 +1,353 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	authhttp "github.com/bsv-blockchain/go-sdk/auth/clients/authhttp"
+)
+
+// StorageUploader implements the StorageUploaderInterface
+type StorageUploader struct {
+	baseURL   string              // Base URL of the storage service
+	authFetch *authhttp.AuthFetch // Authenticated HTTP client for API requests
+}
+
+// NewStorageUploader creates a new uploader instance
+func NewStorageUploader(config UploaderConfig) (*StorageUploader, error) {
+	if config.StorageURL == "" {
+		return nil, errors.New("storage URL is required")
+	}
+	if config.Wallet == nil {
+		return nil, errors.New("wallet is required for authentication")
+	}
+
+	// Create auth fetch client
+	authClient := authhttp.New(config.Wallet, nil, nil)
+
+	return &StorageUploader{
+		baseURL:   config.StorageURL,
+		authFetch: authClient,
+	}, nil
+}
+
+// uploadInfo is used to parse the response from the /upload endpoint
+type uploadInfo struct {
+	Status          string            `json:"status"`
+	UploadURL       string            `json:"uploadURL"`
+	RequiredHeaders map[string]string `json:"requiredHeaders"`
+	Amount          *int64            `json:"amount,omitempty"`
+}
+
+// getUploadInfo requests information from the server to upload a file
+func (u *StorageUploader) getUploadInfo(ctx context.Context, fileSize int, retentionPeriod int) (*uploadInfo, error) {
+	url := fmt.Sprintf("%s/upload", u.baseURL)
```

Comment:
Maybe `uploadURL` so doesn't collide with imported package (also applies to `url` variable in `RenewFile` and `ListUploads`)